app/uk/gov/hmrc/cardpaymentfrontend/models/openbanking/OriginSpecificSessionData.scala

```diff
@@ -344,15 +344,12 @@ final case class VcVatOtherSessionData(vrn: Vrn, vatChargeReference: VatChargeRe
   def searchTag: SearchTag = SearchTag(vatChargeReference.reference)
 }
 
-final case class PfVatSessionData(vrn: Option[Vrn], chargeRef: Option[XRef14Char], returnUrl: Option[Url] = None) extends VatSessionData(PfVat) {
-  private def vatReference: Option[Reference] = vrn.map(ReferenceMaker.makeVatReference)
-  private def chargeReference: Option[Reference] = chargeRef.map(ReferenceMaker.makeXRef14Char)
-
-  def paymentReference: Reference = (vatReference, chargeReference) match {
-    case (Some(vatRef), _) => vatRef
-    case (_, Some(ref))    => ref
-    case _                 => throw new IllegalStateException("[OriginSpecificData][PfVatSessionData] Unable to set paymentReference for PfVatSessionData")
-  }
+final case class PfVatSessionData(vrn: Vrn, chargeRef: Option[XRef14Char], returnUrl: Option[Url] = None) extends VatSessionData(PfVat) {
+  //try and use charge reference as reference, if not provided, use alcoholDutyReference instead.
```

Comment:
the above which has been deleted was lifted and shifted from open-banking/pay-frontend, we shouldn't change it, sorry. Vrn is technically optional as they can enter an xref instead